### PR TITLE
Update to odp v1.35.0.0 level

### PR DESCRIPTION
**Pull request by JereLeppanen**
- Handle deprecations
- Fix a bunch of gcc-10 and other warnings
- Add missing term calls to tests

v2:
- Add commit `Update ODP repository links` (Matias)
- Add commit `timer: Use odp_timer_pool_param_init() instead of memset()` (Matias)
- Add commit `readme: Remove references to travis-ci`
- Add commit `ipsec: If op mode is async, check whether scheduled queues are supported` (Janne)
- Add commit `Remove ThunderX specific documentation`

v3:
- Add review tags.

### CODE REVIEW

**Comment by bogdanPricope** [comment]
Cool.
What about: 
- moving all those globals to shared memory (for multi-process scenarios)
- moving all odp_init_xxx()/odp_term_xxx() under ofp_init_xxx()/ofp_term_xxx() calls (to simplify the code)

**Comment by JereLeppanen** [comment]
@bogdanPricope Possible improvements, but probably not in this PR. ODP and OFP init and term functions have always been separate, and calling the ODP init functions from OFP init functions would probably necessitate passing the parameters through, which would be a bit awkward.

**Comment by bogdanPricope** [comment]
@JereLeppanen I agree that those improvements are not for this PR. I did those and many others in 200+ commits on top OFP (including DHCP and now DNS support).
For ODP/OFP init functions my approach was hybrid: caller either provides an ODP instance (configured with ODP init functions, etc.) or a NFP_ODP_INSTANCE_INVALID value (btw ODP_INSTANCE_INVALID is not defined) and ofp will create and manage the ODP instance with some default params.

Btw, what are your plans with OFP? As far I can tell, for whatever you are using it, I can give you a better version with a richer feature set and it will cost less then develop it yourself.

File: src/ofp_timer.c [hunk 1]
```diff
@@ -131,7 +131,7 @@ static int ofp_timer_create_queues(odp_schedule_group_t sched_group)
 
 	odp_queue_param_init(&param);
 	param.type = ODP_QUEUE_TYPE_SCHED;
-	param.sched.prio  = ODP_SCHED_PRIO_DEFAULT;
+	param.sched.prio  = odp_schedule_default_prio();
```
Comment (MatiasElo) [inline src/ofp_timer.c:134]: New `odp_timer_pool_param_init()` function should also be used.

File: README.md [hunk 2]
```diff
@@ -72,7 +72,7 @@ Download and build OpenDataPlane (ODP) library:
 
     git clone https://git.linaro.org/lng/odp.git
     cd odp
-    git checkout v1.19.0.2
+    git checkout v1.35.0.0
```
Comment (MatiasElo) [inline README.md:75]: Old Linaro URL used above. There seems to be several of these old links in the code base. Could be fixed in a separate commit.

**Comment by JannePeltonen** [comment]
Now that ODP does not necessarily support scheduled queues as IPsec destination queues, ofp_ipsec_init_global() should probably set max_num_sa to zero if inbound_op_mode or outbound_op_mode is async and the queue_type_sched ipsec capability is not set in ODP.

**Review by MatiasElo — APPROVED** [review]

**Pull request merged**